protocol/communities/manager.go

```diff
@@ -1530,6 +1609,164 @@ func (m *Manager) IsSeedingHistoryArchiveTorrent(communityID types.HexBytes) boo
 	return ok && torrent.Seeding()
 }
 
+func (m *Manager) DownloadHistoryArchivesByMagnetlink(communityID types.HexBytes, magnetlink string) ([]string, error) {
+
+	id := communityID.String()
+	ml, err := metainfo.ParseMagnetUri(magnetlink)
+	if err != nil {
+		return nil, err
+	}
+
+	m.archiveLogger.Debug("adding torrent via magnetlink for community", zap.String("id", id), zap.String("magnetlink", magnetlink))
+	torrent, err := m.torrentClient.AddMagnet(magnetlink)
+	if err != nil {
+		return nil, err
+	}
+	m.torrentTasks[id] = ml.InfoHash
+
+	select {
+	case <-time.After(20 * time.Second):
+		return nil, errors.New("torrent has timed out")
+	case <-torrent.GotInfo():
+		files := torrent.Files()
+
+		i, ok := findIndexFile(files)
+		if !ok {
+			// We're dealing with a malformed torrent, so don't do anything
+			return nil, errors.New("malformed torrent data")
+		}
+
+		indexFile := files[i]
+		indexFile.Download()
+		m.archiveLogger.Debug("downloading history archive index")
+
+		ticker := time.NewTicker(200 * time.Millisecond)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
```

Comment:
Demake the busy loop less busy, we're now using a 200 ms ticker, which should be more than enough.